deepchem/models/logger.py

```diff
@@ -0,0 +1,2 @@
+class Logger(object):
+    pass
```

Comment:
Logger needs to provide all the methods that subclasses implement.  It defines the API, and subclasses provide implementations.  In the parent class, all the methods should just throw NotImplementedError.

Reply:
Yup, will fill this in!

I'll add these:
```
log_batch
log_epoch
log_values
setup
finish
```
Not sure about `save_checkpoint()` , that might be something that individual loggers can define?